playbooks/robusta_playbooks/krr.py

```diff
@@ -141,13 +143,42 @@ def krr_scan(event: ExecutionBaseEvent, params: KRRParams):
     Displays a KRR scan report.
     """
 
+    headers = PrometheusAuthorization.get_authorization_headers(params)
+
+    python_command = f"python krr.py {params.strategy} {params.args_sanitized} -p {params.prometheus_url} -q -f json "
+
+    auth_header = headers["Authorization"] if "Authorization" in headers else ""
+
+    # adding env var of auth token from Secret
+    env_var = None
+    secret = None
+    if auth_header:
+        krr_secret_name = "krr-auth-secret"
```

Comment:
I think this may cause issues if there are multiple krr scans in parallel 
The secret creation will fail, or deleted before used etc.

Can we do the following:
For each krr scan create a unique job name
Create the job (where the container is dependent on the secret)
Create the secret with the job as owner.

When the job is done and deleted, the secret will be deleted with it.
The container will not start until the secret is created.
This also removes the need for secret read/patch permission. Only write is required. 
WDYT?

Reply:
when i tested it I kept getting a CreateContainerConfigError and the pod wouldnt run but I think I found the issue so ill try it

Reply:
but then you have the issue if the secret isnt sucessfully created than the job will exist forever

Reply:
I tested it manually. The container had indeed the `CreateContainerConfigError` error, but once I created the secret it ran and finished successfully

Reply:
The GC didnt delete the secret when the Job was the owner but did delete it when the Job Pod was the owner